provisioning/service/src/Twin/InitialTwinPropertyCollection.cs

```diff
@@ -205,6 +205,12 @@ public IEnumerator GetEnumerator()
             }
         }
 
+        /// <inheritdoc/>
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
```

Comment:
Is this method just calling itself?

Reply:
And why do we need two "GetEnumerator()" functions in this class?

Reply:
It seems that way, isn't it!

It's calling https://github.com/Azure/azure-iot-sdk-csharp/blob/78138673c18535408c0321d7aab82e0acbce8f02/provisioning/service/src/Twin/InitialTwinPropertyCollection.cs#L195-L206

Reply:
We need this one because InitialTwinPropertyCollection an `IEnumerator GetEnumerator()` iterator.

Reply:
The interface `IEnumerable<T>` requires the method `IEnumerator<T> GetEnumerator();` to be implemented. At the same time, `IEnumerable<T>` implements `IEnumerable`, which requires the method `IEnumerator GetEnumerator();` to be implemented.